### PR TITLE
Feature/56115 HDF soft deletion for service manager

### PR DESCRIPTION
**Pull request by GuyHarwood**
- allow service manager role to soft delete HDF signings
- 'slim' first pass version with minimal functionality

### CODE REVIEW

File: admin/services/data-access/headteacher-declaration.data.service.js [hunk 1]
```diff
@@ -143,4 +147,46 @@ headteacherDeclarationDataService.sqlFindPupilsBlockingHdf = async (schoolId) =>
...
+ * soft delete the hdf entry
+ * @param schoolId
+ * @param userId
+ * @return {Promise<any>}
+ */
+headteacherDeclarationDataService.sqlUndoSoftDeleteHdfEntry = async (schoolId, userId) => {
+  const sql = `
+    UPDATE [mtc_admin].[hdf]
+    SET isDeleted = 0, deletedAt = NULL, deletedBy_userId = NULL
+    WHERE school_id = @schoolId
```
Comment (jon-shipley) [inline admin/services/data-access/headteacher-declaration.data.service.js:181]: Should this also have the WHERE clause ' WHERE isDeleted = 1' ?

File: db/migrations/schema-objects/20230609145304.do.hdf-deletion-columns.sql [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      ADD [isDeleted] BIT CONSTRAINT [DEFAULT_hdf_isDeleted] DEFAULT 0 NOT NULL;
+  END
+
+IF NOT EXISTS(SELECT *
+                FROM sys.columns
+               WHERE object_ID = object_id('[mtc_admin].[hdf]')
+                 AND col_name(object_ID, column_Id) = 'deletedAt')
+  BEGIN
+  ALTER TABLE [mtc_admin].[hdf]
+      ADD [deletedAt] DATETIME2 (3) NULL;
```
Comment (jon-shipley) [inline db/migrations/schema-objects/20230609145304.do.hdf-deletion-columns.sql:16]: Other deletedAt cols use DATETIMEOFFSET e.g. pupilRestart

**Review by jon-shipley — APPROVED** [review]
LGTM - one suggestion made. 👍

File: db/migrations/schema-objects/20230609145304.do.hdf-deletion-columns.sql (hunk 2 above)
Comment (GuyHarwood) [reply db/migrations/schema-objects/20230609145304.do.hdf-deletion-columns.sql:16]: good catch, gotta be careful with that copilot autocomplete!

**Review by jon-shipley — APPROVED** [review]
LGTM

**Pull request merged**